test/zone.spec.js

```diff
@@ -11,6 +11,24 @@ describe('Zone', function () {
     expect(zone.fork().$id).toBeGreaterThan(zone.$id);
   });
 
+  it('should be possible to override the auto-generated id', function () {
+    zone.fork({'$id': 'test'}).run(function () {
+      expect(zone.$id).toEqual('test');
+    });
+  });
+
+  it('should return the full path', function () {
+    Zone.nextId = 2;
+
+    zone.fork().run(function () {
+      zone.fork({'$id': 'foo'}).run(function () {
```

Comment:
why not keep the number as is. I think it's useful to have a sequential number that helps us determine when (relatively speaking) a zone was created.

also exposing $id as a public api seems like an abstraction leak.

instead, I propose that there is a property called "name" (possibly confusing because of window.name), "label", "zoneName" or "zoneLabel" on a zone, which by default is an empty string, but can be overridden just like any `setTimeout` or other hooks.

path could then return a combo of $id and this label property, so you'd get `"root#1 > #2 > angularTurnZone#3 -> #4 > #5"`